browser/ui/views/brave_shields/cookie_list_opt_in_browsertest.cc

```diff
@@ -0,0 +1,176 @@
+ public:
+  static inline constexpr char kRegionalAdBlockComponentTestId[] =
+      "lfgnenkkneohplacnfabidofpgcdpofm";
+
+  static inline constexpr char kRegionalAdBlockComponentTest64PublicKey[] =
+      "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAqNcRp37CBZCXe1vbmXyobycPxHyE"
+      "KNIgNl6p0XBBxtcZcQOijpY70GjCRzgCL7m1+FBo4MR3FXLiF2aPn/"
+      "QsUR8t7+zfw3XzBVos4Ssexkqpd4/"
+      "4ciASwTXpbuyFOq4Z5dcgJ1afeT9Zj5bmh4ekLpgJ1NzVwCMhEKk6cmSKIaGVo5EEydtlor2"
+      "nkUJrSFuZA6tYZ++"
+      "4BOfhhCtzrvXTZjg7mTlB6ca21NL4oLwtqvJMtF8ddoumh619BB5wOqxLzntC/"
+      "oWyOxf00V5HDC7e/"
+      "DRj9J8jLRFLd4EQUO4Mk+kG3MNy0ph9cqdw6zFR7a2H3LGkl4ejsifM1mUDuJL0cwIDAQAB";
+
+  CookieListOptInBrowserTest() {
+    scoped_feature_list_.InitWithFeatures(
+        {features::kBraveAdblockCookieListOptIn}, {});
+  }
+
+  ~CookieListOptInBrowserTest() override = default;
+
+ protected:
+  void SetUpOnMainThread() override {
+    StartServer();
+    InitializeFilterLists();
+  }
+
+  GURL GetTestURL(std::string host) {
+    return embedded_test_server()->GetURL(host, "/simple.html");
+  }
+
+  auto* regional_service_manager() {
+    return g_brave_browser_process->ad_block_service()
+        ->regional_service_manager();
+  }
+
+  content::WebContents* GetBubbleWebContents() {
+    auto* host = CookieListOptInBubbleHost::FromBrowser(browser());
+    return host ? host->GetBubbleWebContentsForTesting() : nullptr;
+  }
+
+ private:
+  void StartServer() {
+    using net::test_server::BasicHttpResponse;
+    using net::test_server::HttpRequest;
+    using net::test_server::HttpResponse;
+
+    host_resolver()->AddRule("*", "127.0.0.1");
+
+    embedded_test_server()->RegisterRequestHandler(
+        base::BindRepeating([](const HttpRequest& request) {
+          auto response = std::make_unique<BasicHttpResponse>();
+          response->set_code(net::HTTP_OK);
+          response->set_content("hello");
+          response->set_content_type("text/plain");
+          return std::unique_ptr<HttpResponse>(std::move(response));
+        }));
+
+    ASSERT_TRUE(embedded_test_server()->Start());
+  }
+
+  void InitializeFilterLists() {
+    std::vector regional_catalog = {adblock::FilterList(
+        kCookieListUuid,
+        "https://secure.fanboy.co.nz/fanboy-cookiemonster_ubo.txt",
+        "Easylist-Cookie List - Filter Obtrusive Cookie Notices", {},
+        "https://forums.lanik.us/", kRegionalAdBlockComponentTestId,
+        kRegionalAdBlockComponentTest64PublicKey,
+        "Removes obtrusive cookie law notices")};
+    regional_service_manager()->SetRegionalCatalog(regional_catalog);
+  }
+
+  base::test::ScopedFeatureList scoped_feature_list_;
+};
+
+IN_PROC_BROWSER_TEST_F(CookieListOptInBrowserTest, EnableFromBubble) {
+  auto* bubble_host = CookieListOptInBubbleHost::FromBrowser(browser());
+  ASSERT_TRUE(bubble_host);
+  bubble_host->set_allow_in_background_for_testing(true);
+
+  EXPECT_TRUE(ui_test_utils::NavigateToURL(browser(), GetTestURL("a.com")));
```

Comment:
for existing users this should show up on the first open tab regardless of url so we shouldn't have to navigate first to trigger it https://bravesoftware.slack.com/archives/C07HD31EZ/p1658868232784399?thread_ts=1657224065.754619&cid=C07HD31EZ

Reply:
Good point - in that version the tests were doing setup after the initial tab was displayed. I've updated the tests to perform setup in `PreRunTestOnMainThread`.